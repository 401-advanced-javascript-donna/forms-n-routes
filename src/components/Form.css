.Form {
  display: flex;
  flex-direction: column;
  align-items: center;

  h2 {
    text-transform: uppercase;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1em;
    margin-top: 2em;
    margin-bottom: .75em;
  }

  input {
    line-height: 20px;
    width: 300px;
    margin-bottom: 10px;
  }
  
  button {
    text-transform: uppercase;
    padding: 5px;
    background-color: #402F3F;
    font-weight: 700;
    color: #A0B890;
    font-size: 16px;
    width: 150px;
    cursor: pointer;
  }

  button:focus{
    outline: none;
  }
}