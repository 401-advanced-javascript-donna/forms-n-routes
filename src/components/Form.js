import React from 'react';
import PropTypes from 'prop-types';

const Form = ({ form, handleSubmit, handleChange }) => {

  return (
    <form onSubmit={handleSubmit}>
      <h2>Find a Character</h2>
      <input type="text" name="form" value={form} onChange={handleChange}></input>
      <button>Form</button>
    </form>
  );
};

Form.propTypes = {
  form: PropTypes.string.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired
};


export default Form;
