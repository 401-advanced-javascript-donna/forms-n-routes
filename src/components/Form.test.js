import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form component', () => {
  it('should render Form', () => {
    const wrapper = shallow(<Form
      search="test search"
      handleChange={() => { }}
      handleSubmit={() => { }}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
