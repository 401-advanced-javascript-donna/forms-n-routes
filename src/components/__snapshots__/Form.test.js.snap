// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component should render Form 1`] = `
<form
  className="Form"
  onSubmit={[Function]}
>
  <h2>
    Find a Character
  </h2>
  <input
    name="search"
    onChange={[Function]}
    type="text"
    value="test search"
  />
  <button>
    Search
  </button>
</form>
`;
